ProcessLib/LIE/HydroMechanics/HydroMechanicsProcess.cpp

```diff
@@ -100,63 +126,68 @@ void HydroMechanicsProcess<GlobalDim>::constructDofTable()
     // prepare mesh subsets to define DoFs
     //------------------------------------------------------------
     // for extrapolation
-    _mesh_subset_all_nodes.reset(new MeshLib::MeshSubset(_mesh, &_mesh.getNodes()));
+    _mesh_subset_all_nodes.reset(
+        new MeshLib::MeshSubset(_mesh, &_mesh.getNodes()));
     // pressure
-    _mesh_nodes_p = MeshLib::getBaseNodes(_mesh.getElements());
+    _mesh_nodes_p = MeshLib::getBaseNodes(
+        _process_data.p_element_status->getActiveElements());
     _mesh_subset_nodes_p.reset(new MeshLib::MeshSubset(_mesh, &_mesh_nodes_p));
     // regular u
-    _mesh_subset_matrix_nodes.reset(new MeshLib::MeshSubset(_mesh, &_mesh.getNodes()));
+    _mesh_subset_matrix_nodes.reset(
+        new MeshLib::MeshSubset(_mesh, &_mesh.getNodes()));
     if (!_vec_fracture_nodes.empty())
     {
         // u jump
-        _mesh_subset_fracture_nodes.reset(new MeshLib::MeshSubset(_mesh, &_vec_fracture_nodes));
+        _mesh_subset_fracture_nodes.reset(
+            new MeshLib::MeshSubset(_mesh, &_vec_fracture_nodes));
     }
 
     // Collect the mesh subsets in a vector.
     std::vector<std::unique_ptr<MeshLib::MeshSubsets>> all_mesh_subsets;
     std::vector<unsigned> vec_n_components;
-    std::vector<std::vector<MeshLib::Element*>const*> vec_var_elements;
+    std::vector<std::vector<MeshLib::Element*> const*> vec_var_elements;
     // pressure
     vec_n_components.push_back(1);
     all_mesh_subsets.push_back(std::unique_ptr<MeshLib::MeshSubsets>{
         new MeshLib::MeshSubsets{_mesh_subset_nodes_p.get()}});
-    vec_var_elements.push_back(&_mesh.getElements());
+    if (!_process_data.deactivate_matrix_in_flow)
+    {
+        vec_var_elements.push_back(&_mesh.getElements());
+    }
+    else
+    {
+        // TODO set elements including active nodes for pressure.
```

Comment:
This is because matrix elements connecting to a fracture still include nodes having active DoF for pressure and should be included in a vector of elements for assembly.